app/assets/javascripts/organisation-list-filter.js

```diff
@@ -8,6 +8,8 @@
     init: function() {
       // Form should only appear if the JS is working
       $('[data-filter="form"]').addClass('filter-organisations-list__form--active');
+      var $results = $('<div/>').addClass('filter-organisations-list__results govuk-heading-m').attr('id', 'search-results').attr('aria-live', 'polite').html(window.GOVUK.filter.countInitialDepartments() + ' results found');
```

Comment:
Couple of things:

- It appears that this code is rendering a div with classes that can contain the results list. Could we render this in the markup itself instead then just grab it via a selector?
- Could we vanillaify the new js? I reckon most of these could stay as single lines, mostly just replacing `$`'s with `querySelector`.

Reply:
This div holds the result count from the user's search and search isn't available when JS is disabled. Since this element is specifically related to search I didn't think it should appear in the markup, as it would appear even when JS is disabled.

I could vanilla this JS, but as the whole of the rest of the file uses jQuery already I think that's outside the scope of this PR.

Reply:
I take your point on the search box however whilst it doesn't appear without js it's still stored in the markup, just hidden. Could we hide the results count in the same way? I'm thinking that it'll be easier for a future dev to find and minimise js processing.

Willing to fold on vanilla js.

Reply:
Oh I see... you're saying that because the search box is in the markup but hidden, the search results should follow the same pattern?

I get that, but the only reason the search box isn't rendered with JS as well is because it's a component. I'd rather try to have as many JS-only elements rendered with JS than make all JS only elements one way or the other.

Reply:
I don't agree that we should be rendering anything via js if we can avoid it. This ensure we can separate template from functionality, making the code easier to interpret and  minimising the work needed to be done by the js.

I appreciate that this is my opinion though and don't want to perpetuate a stalemate more than I'm already starting to. If there's a case for this, like if it's a common pattern in our apps or there's a performance hit to rendering it in the html etc, then I'll let it go.

Reply:
@injms

Reply:
🐱 🐦 time.

I think that the element is fine to be added by the JavaScript - but I would suggest changing the selector that it uses to add the new element (`#organisations_search_results`) to a `js-` hook. I've added this as a comment.

In a previous [GOV.UK Frontenders meeting](https://docs.google.com/document/d/1RsB7WQBkt3mH3yOD46VMNw8ZI9VTa04jyS0HXQfeYJI/edit?usp=sharing) there was a decision to not use jQuery when writing JavaScript - and it was fine if that meant mixing JavaScript and jQuery together. It'd be good to see that in this pull request - but not a blocker to it being merged as we're under tight time constraints.

Reply:
Thanks @injms. I've updated to use a `js-` hook as suggested. I've also rewritten the code to use vanilla JS @owenatgov except for one line where I'm using jQuery's `prepend` method - I tried switching to an alternative but it started throwing up weird test failures so I'm quitting while I'm ahead on that, sorry.

I've also changed the code slightly further to remove an element at the bottom of the page that I've only just noticed, that showed with a message when there were no search results, which is redundant now.

Reply:
Will hold off merging for a bit if you both want to take a second look.